tests/components/FilterSearch.test.tsx

```diff
@@ -451,4 +451,39 @@ it('clears input when old filters are removed', async () => {
   await waitFor(() => {
     expect(searchBarElement).toHaveValue('');
   });
+});
+
+it('toggling the dropdown does not change selected filters', async () => {
+  const setFilterOption = jest.spyOn(SearchHeadless.prototype, 'setFilterOption');
+  const executeFilterSearch = jest
+    .spyOn(SearchHeadless.prototype, 'executeFilterSearch')
+    .mockResolvedValue(labeledFilterSearchResponse);
+
+  render(
+    <div>
+      <SearchHeadlessContext.Provider value={generateMockedHeadless(mockedState)}>
+        <FilterSearch searchFields={searchFieldsProp} />
+      </SearchHeadlessContext.Provider>
+      <div>external div</div>
+    </div>
+  );
+
+  const searchBarElement = screen.getByRole('textbox');
+  const externalDiv = screen.getByText('external div');
+  userEvent.type(searchBarElement, 'first name 1');
+  expect(await screen.findByRole('textbox')).toHaveDisplayValue('first name 1');
+  userEvent.type(searchBarElement, '{enter}');
+
+  await waitFor(() => {
+    expect(executeFilterSearch).toHaveBeenCalled();
+  });
+
+  await waitFor(() => {
+    expect(setFilterOption).toBeCalledTimes(1);
+  });
+
```

Comment:
is it possible to move this boilerplate to a function, and share it with the other tests that do the same thing? That 1. helps reduce repeated code and 2. helps describe what this block of code is doing by giving it a name (self documenting code)

Reply:
I highlighted the lines when writing this comment but it looks like it didn't come through, by boilerplate I mean from the start of the test down to before we click on searchBarElement

Reply:
I think all of the tests do something slightly different; like in this test I'm counting how many times `executeFilterSearch` is called and in the other one it's checking what it's been called with, and so on. The rendering is also a bit different in each; in this one I need an external div to click out into. lmk if there's a specific block of code you think I can put into a function- I don't really see one though lol